src/LinuxCaptureService/include/LinuxCaptureService/LinuxCaptureService.h

```diff
@@ -48,8 +50,8 @@ class LinuxCaptureService final : public orbit_capture_service::CaptureService {
   //   CaptureService::WaitForStopCaptureRequestFromClient);
   // - MemoryWatchdog calls OnThresholdExceeded.
   [[nodiscard]] StopCaptureReason WaitForStopCaptureRequestOrMemoryThresholdExceeded(
-      grpc::ServerReaderWriter<orbit_grpc_protos::CaptureResponse,
-                               orbit_grpc_protos::CaptureRequest>* reader_writer);
+      const std::shared_ptr<orbit_capture_service::StartStopCaptureRequestWaiter>&
```

Comment:
Why is that a shared pointer here?

Reply:
Because it spins up a background thread that needs access to the waiter. I believe that's proper usage of a shared_ptr because we can't fully control the lifetime of that thread.

Reply:
Makes sense. Thanks!